File: 0002-add-two-numbers/0002-add-two-numbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        num1 = ""
        while l1:
            num1 += str(l1.val)
            l1 = l1.next
            
        num2 = ""
        while l2:
            num2 += str(l2.val)
            l2 = l2.next
        
        num1 = num1[::-1]
        num2 = num2[::-1]
        res = int(num1) + int(num2)
        test = res % 10        

        node = ListNode(test, None)
        
        res = (res//10)
        
        ans = node
        
        while res != 0:
            node.next = ListNode(res%10, None)

            res = res // 10
            node = node.next
            
        return ans
        

File: 0002-add-two-numbers/0002-add-two-numbers.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        ListNode answer = new ListNode(0);
        ListNode head = answer;
        
        int nTot = 0;
        while(l1 != null || l2 != null){
            int tot = 0;
            tot += nTot;
            if(l1 != null){
                tot += l1.val;    
            }
            if(l2 != null){
                tot += l2.val;                
            }
            
            if (tot >=10){
                tot -= 10;
                nTot = 1;
            }else{
                nTot = 0;
            }
            answer.val = tot;
            if(l1 != null){
                l1 = l1.next;    
            }
            if(l2 != null){
                l2 = l2.next;    
            }
            
            
            if(l1 != null || l2 != null){
                answer.next = new ListNode(0);    
                answer = answer.next;
            }
        }
        if(nTot == 1){
            answer.next = new ListNode(1);
        }
        return head;
    }
}